plugins/org.jboss.tools.openshift.io.core/src/main/java/org/jboss/tools/openshift/io/internal/core/OpenShiftIOCoreActivator.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.io.internal.core;
+
+import org.jboss.tools.foundation.core.plugin.BaseCorePlugin;
+
+public class OpenShiftIOCoreActivator extends BaseCorePlugin {
+
+	private static OpenShiftIOCoreActivator INSTANCE = null;
+	
+	public static final String PLUGIN_ID = "org.jboss.tools.openshift.io.core";
+	
+	public OpenShiftIOCoreActivator() {
+		super();
+		INSTANCE = this;
+	}
+
+	public static OpenShiftIOCoreActivator getDefault() {
+		return INSTANCE;
```

Comment:
potential NPE, e.g. on line 31. If constructor hasn't been called yet, the `INSTANCE` is `null`

Reply:
Not in the OSGI world